src/SelfService/Application/ConfigurationLevelService.cs

```diff
@@ -61,12 +62,27 @@ public void AddMetric(ConfigurationLevelDetail detail)
 
 public class ConfigurationLevelService : IConfigurationLevelService
 {
+    private readonly IKafkaTopicRepository _kafkaTopicRepository;
+    private readonly IMessageContractRepository _messageContractRepository;
+    private readonly ICapabilityRepository _capabilityRepository;
+
+    public ConfigurationLevelService(
+        IKafkaTopicRepository kafkaTopicRepository,
+        IMessageContractRepository messageContractRepository,
+        ICapabilityRepository capabilityRepository
+    )
+    {
+        _kafkaTopicRepository = kafkaTopicRepository;
+        _messageContractRepository = messageContractRepository;
+        _capabilityRepository = capabilityRepository;
+    }
+
     public async Task<ConfigurationLevelInfo> ComputeConfigurationLevel(CapabilityId capabilityId)
     {
         var configLevelInfo = new ConfigurationLevelInfo();
         configLevelInfo.AddMetric(
             new ConfigurationLevelDetail(
-                await GetKafkaTopicConfigurationLevel(),
+                await GetKafkaTopicConfigurationLevel(_kafkaTopicRepository, capabilityId, _messageContractRepository),
```

Comment:
you don't have to pass the repositories here, I believe

Reply:
But... I wrote the function to expect those arguments? :thinking: 
You mean use them as attributes of the class instead?

Reply:
ok I see. Done